rest/tests/test_rest_authorization.py

```diff
@@ -0,0 +1,476 @@
+"""
+Tests for REST authorization
+"""
+
+from __future__ import unicode_literals
+
+from rest_framework.status import (
+    HTTP_400_BAD_REQUEST,
+    HTTP_403_FORBIDDEN,
+    HTTP_405_METHOD_NOT_ALLOWED,
+)
+from django.contrib.auth.models import User, Permission
+
+from .base import RESTTestCase
+from roles.api import assign_user_to_repo_group
+from roles.permissions import GroupTypes
+
+
+class TestRestAuthorization(RESTTestCase):
+    """
+    Tests for REST authorization
+    """
+
+    def setUp(self):
+        super(TestRestAuthorization, self).setUp()
+
+        # add_repo_user is another user with add_repo permission but who
+        # does not have access to self.repo
+        self.add_repo_user = User.objects.create_user(
+            username="creator_user", password=self.PASSWORD
+        )
+        add_repo_perm = Permission.objects.get(codename=self.ADD_REPO_PERM)
+        self.add_repo_user.user_permissions.add(add_repo_perm)
+
+        # curator_user doesn't have add_repo but have view_repo permission on
+        # self.repo
+        self.curator_user = User.objects.create_user(
+            username="curator_user", password=self.PASSWORD
+        )
+        assign_user_to_repo_group(
+            self.curator_user, self.repo, GroupTypes.REPO_CURATOR)
+
+        # author_user doesn't have manage_taxonomy permission
+        self.author_user = User.objects.create_user(
+            username="author_user", password=self.PASSWORD
+        )
+        assign_user_to_repo_group(
+            self.author_user, self.repo, GroupTypes.REPO_AUTHOR
+        )
+
+    def test_repository_create(self):
+        """Test create repository authorization"""
+        # switch to add_repo_user
+        self.logout()
+        self.login(self.add_repo_user.username)
+
+        # add_repo_user creates a repository
+        new_repo_slug = self.create_repository()['slug']
+
+        self.get_repository(new_repo_slug)
+        self.get_repository(self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        repos = self.get_repositories()
+        self.assertEqual(1, repos['count'])
+
+        # duplicate repository, now that we assigned another repo
+        # with same name to user
+        self.create_repository(expected_status=HTTP_400_BAD_REQUEST)
+
+        # switch to curator_user
+        self.logout()
+        self.login(self.curator_user.username)
+
+        self.create_repository(expected_status=HTTP_403_FORBIDDEN)
+
+        self.create_repository({
```

Comment:
Is this test redundant, since the first proves a 403 for that call for that user?